R/mergeBenchmarkResults.R

```diff
@@ -0,0 +1,78 @@
+#' @title Merge different BenchmarkResult objects.
+#' @description Combines the \code{\link{BenchmarkResult}} objects that were performed 
+#'   with different learners and tasks. 
+#'   This can be helpful if you, e.g. forgot to run one learner on the set of tasks you used.
+#' @param ... [\code{\link{BenchmarkResult}}]\cr 
+#'   \code{BenchmarkResult} objects that should be merged.
+#' @return \code{\link{BenchmarkResult}}
+#' @details Note that if you want to merge several \code{\link{BenchmarkResult}}
+#'   objects you must ensure that all possible learner and task combinations will be
+#'   contained in the return object.\cr
+#'   Furthermore all given objects must have been calculated on the same
+#'   set of measures.
+#' @export
+mergeBenchmarkResults = function(...) {
+  set = list(...)
+  for (i in 1:length(set)) {
+    assertClass(set[[i]], "BenchmarkResult")
+  }
+  # Check if bmrs optimized on same measure
+  measures = lapply(set, function(bmr) {
+    getBMRMeasureIds(bmr)
+  })
+  if (length(unique(measures)) > 1L)
```

Comment:
check seems wrong
maybe better: if `length(Reduce("intersect", lapply(set, getBMRMeasureIds)))>0` then we have at least one measure among all bmr's that is the same.